go/processor/floria/run_context.go

```diff
@@ -170,6 +172,10 @@ func transferValue(
 		return fmt.Errorf("insufficient balance: %v < %v", senderBalance, value)
 	}
 
+	if sender == recipient {
```

Comment:
Move this either to #755 or remove the modification of `transferValue` in #755 and move the test for this function into this PR.

Reply:
I moved all changed related to value transfers to this PR and updated the unit test.